aptos-move/framework/supra-framework/sources/dora_committee.move

```diff
@@ -0,0 +1,1043 @@
+    public fun get_committee_id(
+        com_store_addr: address,
+        node_address: address
+    ): u64 acquires CommitteeInfoStore {
+        let committee_store = borrow_global<CommitteeInfoStore>(com_store_addr);
+        *simple_map::borrow(&committee_store.node_to_committee_map, &node_address)
+    }
+
+    #[view]
+    /// Get the node's information
+    public fun get_node_info(
+        com_store_addr: address,
+        id: u64,
+        node_address: address
+    ): NodeData acquires CommitteeInfoStore {
+        let committee_store = borrow_global<CommitteeInfoStore>(com_store_addr);
+        let committee = simple_map::borrow(&committee_store.committee_map, &id);
+        let (addrs, nodes) = simple_map::to_vec_pair(committee.map);
+        let (flag, index) = vector::index_of(&addrs, &node_address);
+        assert!(flag, error::invalid_argument(NODE_NOT_FOUND));
+        let node_info = vector::borrow(&nodes, index);
+
+        NodeData {
+            operator: node_address,
+            ip_public_address: node_info.ip_public_address,
+            node_public_key: node_info.node_public_key,
+            network_public_key: node_info.network_public_key,
+            cg_public_key: node_info.cg_public_key,
+            network_port: node_info.network_port,
+            rpc_port: node_info.rpc_port,
+        }
+    }
+
+    #[view]
+    /// Get the committee's id for a single node
+    public fun get_committee_id_for_node(
+        com_store_addr: address,
+        node_address: address
+    ): u64 acquires CommitteeInfoStore {
+        let committee_store = borrow_global<CommitteeInfoStore>(com_store_addr);
+        let id = *simple_map::borrow(&committee_store.node_to_committee_map, &node_address);
+        id
+    }
+
+    #[view]
+    /// Get a tuple of the node itself and node peers vector for a single node
+    public fun get_peers_for_node(
+        com_store_addr: address,
+        node_address: address
+    ): (NodeData, vector<NodeData>) acquires CommitteeInfoStore {
+        let committee_id = get_committee_id_for_node(com_store_addr, node_address);
+        let this_node = get_node_info(com_store_addr, committee_id, node_address);
+        let (node_info,_) = get_committee_info(com_store_addr, committee_id);
+        let (_, index) = vector::index_of(&node_info, &this_node);
+        let self= vector::remove(&mut node_info, index);
+        (self, node_info)
+    }
+
+    #[view]
+    /// Check if the committee has a valid dkg
+    public fun does_com_have_dkg(com_store_addr: address, com_id: u64): bool acquires CommitteeInfoStore {
+        let committee_store = borrow_global<CommitteeInfoStore>(com_store_addr);
+        let committee = simple_map::borrow(&committee_store.committee_map, &com_id);
+        committee.has_valid_dkg
+    }
+
+    /// Update the dkg flag
+    public fun update_dkg_flag(
+        com_store_addr: address,
+        owner_signer: &signer,
```

Comment:
We can make this function as **entry** function.
Also `&signer` should always be the first parameter, Please update it everywhere.

Reply:
Updated.

Reply:
Also renamed the file.